src/main/java/org/broadinstitute/hellbender/engine/dataflow/datasources/ReferenceShard.java

```diff
@@ -0,0 +1,92 @@
+package org.broadinstitute.hellbender.engine.dataflow.datasources;
+
+import com.google.cloud.dataflow.sdk.coders.*;
+import com.google.cloud.dataflow.sdk.values.KV;
+import htsjdk.samtools.util.Locatable;
+import org.broadinstitute.hellbender.utils.Utils;
+
+import java.io.Serializable;
+
+/**
+ * ReferenceShard is section of the reference genome that's used for sharding work for pairing things with
+ * the reference. The primary use case is pairing reads with the reference.
+ * This class designed to be simple a data-storage class with related static utilities.
+ */
+public class ReferenceShard implements Serializable {
+    private static final long serialVersionUID = 1L;
+
+    private final int shardNumber; // shardNumber is zero-based.
+    private final String contig;
+
+    public static int REFERENCE_SHARD_SIZE = 100000; // This value is likely to be tweaked.
+
+    public ReferenceShard(int shardNumber, String contig) {
+        this.shardNumber = shardNumber;
+        Utils.nonNull(contig);
+        this.contig = contig;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        ReferenceShard that = (ReferenceShard) o;
+
+        if (getShardNumber() != that.getShardNumber()) return false;
+        return getContig().equals(that.getContig());
+
+    }
+
+    @Override
+    public int hashCode() {
+        int result = getShardNumber();
+        result = 31 * result + getContig().hashCode();
+        return result;
+    }
+
+    public String getContig() {
+        return contig;
+    }
+
+    public int getShardNumber() {
+
+        return shardNumber;
+    }
+
+    @Override
+    public String toString() {
+        return "ReferenceShard{" +
+                "shardNumber=" + shardNumber +
+                ", contig='" + contig + '\'' +
+                '}';
+    }
+
+    /**
+     * getShardNumberFromInterval determines which shard the location maps to (using the start position)
```

Comment:
I would suggest instead:
getShardNumberFromInterval returns the ReferenceShard that overlap the read's start position.